### DISCUSSION TITLE
kubernetes cluster monitoring

### DISCUSSION THREAD

**Discussion by piyush-padmarag** [apache/skywalking#10590] [Q&A]
Hi

I have an active k8s cluster with prometheus elastic search kube-state-metrics and all. i configured skywalking to fetch vm metrics of the prometheus vm using open telemetry by giving elastic search details in application.yml of SW . 

Now, i want to get metrics for my k8s cluster as well on ASW. kindly advise me the steps in detail. What changes should i do in config.yml file of OTEL and application.yml of SW. ?

**Comment by wu-sheng** [comment]
Ref doc https://skywalking.apache.org/docs/main/next/en/setup/backend/backend-k8s-monitoring/
The otel configuration file is attached in the docs.

**Reply by piyush-padmarag** [reply]
what things should be changed in the config file ? i am using https://raw.githubusercontent.com/open-telemetry/opentelemetry-collector/main/examples/k8s/otel-config.yaml

asking it as i am getting error while dialing.... connection refused. on each pod

**Reply by piyush-padmarag** [reply]
Hi
kindly provide steps to authorize OAP server (outside the cluster) to authenticate the k8s api server. SW is outside the cluster but in the same resource group as of my cluster

**Reply by wu-sheng** [reply]
I think now we don't have such mechanism inside SkyWalking.
If you want to expose K8s API server out of cluster, you need to search solutions at k8s side.
